plugins/integrations/prometheus/src/main/java/org/apache/cloudstack/metrics/PrometheusExporterImpl.java

```diff
@@ -131,6 +142,24 @@ private void addHostMetrics(final List<Item> metricsList, final long dcId, final
             int isDedicated = (dr != null) ? 1 : 0;
             metricsList.add(new ItemHostIsDedicated(zoneName, zoneUuid, host.getName(), host.getUuid(), host.getPrivateIpAddress(), isDedicated));
 
+            List<String> hostTags = _hostTagsDao.gethostTags(host.getId());
+            String hosttags = StringUtils.join(hostTags, ",");
+            for (String tag : hostTags) {
+                Integer current = totalHosts.get(tag) != null ? totalHosts.get(tag) : 0;
+                totalHosts.put(tag, current + 1);
+            }
+            if (host.getStatus() == Status.Up) {
+                for (String tag : hostTags) {
+                    Integer current = upHosts.get(tag) != null ? upHosts.get(tag) : 0;
+                    upHosts.put(tag, current + 1);
+                }
+            } else if (host.getStatus() == Status.Disconnected || host.getStatus() == Status.Down) {
+                for (String tag : hostTags) {
+                    Integer current = downHosts.get(tag) != null ? downHosts.get(tag) : 0;
+                    downHosts.put(tag, current + 1);
+                }
+            }
+
```

Comment:
addHostMetrics() is already way to long, and the new functionality shouldn't be added to it, so as an example
this functionality seems like it could be extracted and there is some repetition in it that can be generalised as well.
```suggestion
            String hosttags = markTagMaps(totalHosts, upHosts,  downHosts);
```
and then add
```
        private String markTagMaps(Map<String, Integer> totalHosts, Map<String, Integer> upHosts, Map<String, Integer> downHosts) {
            List<String> hostTags = _hostTagsDao.gethostTags(host.getId());
            markTags(hostTags,totalHosts);
            if (host.getStatus() == Status.Up) {
                markTags(hostTags, upHosts);
            } else if (host.getStatus() == Status.Disconnected || host.getStatus() == Status.Down) {
                markTags(hostTags, downHosts);
            }
            return StringUtils.join(hostTags, ",");
        }
```
and
```
  private void markTags(List<String tags, Map<String, Integer> tagMap) {
            for (String tag : tags) {
                Integer current = tagMap.get(tag) != null ? tagMap.get(tag) : 0;
                tagMap.put(tag, current + 1);
            }
        }
```